pkg/config/expand.go

```diff
@@ -525,43 +524,42 @@ func (dg *DeploymentGroup) identifySimpleVariable(yamlReference string) (varRefe
 	return ref, nil
 }
 
-func (ref *modReference) validate(depGroups []DeploymentGroup) error {
-	callingModuleGroupIndex := slices.IndexFunc(depGroups, func(d DeploymentGroup) bool { return d.Name == ref.SourceGroupID })
+func (ref *modReference) validate(depGroups []DeploymentGroup, modToGrp map[string]int) error {
+	callingModuleGroupIndex := slices.IndexFunc(depGroups, func(d DeploymentGroup) bool { return d.Name == ref.FromGroupID })
```

Comment:
Could you just use modToGrp to look up this index? Then you would not have to pass in depGroups at all.

Reply:
Yes, that is vestigial, so to speak. Will address.

Reply:
I take that back, I am not tracking the "FromModuleID" just, "FromGroupID" which requires a search for the actual group ID (not present in `modToGrp`). Given the scale of our blueprints, I have erred on the side of searching our structs vs "custom" objects like the `map[string]int`.

Reply:
Ok, I see. If I were designing from scratch I might want to swap out the value of the modToGrp map  (currently int) to contain both the index and the name. 🤷 . I will let it be. We are getting into the business of creating types to represent blueprint objects, which I generally think is good, but I don't know what all would be involved here.

Feel free to close this and leave as is if you like.